runtime/api/src/main/java/org/qbicc/runtime/Build.java

```diff
@@ -197,7 +197,7 @@ public static boolean isWasm() {
 
         @Fold
         public static boolean isWasi() {
-            return defined(__wasi__);
+            return defined(__wasi__) || defined(__EMSCRIPTEN__);
```

Comment:
Relatedly, you might want to add a WASI condition to `org.qbicc.plugin.core.ConditionEvaluation` as well.